### ISSUE TITLE
Content assist inserts wrong arguments

### ISSUE THREAD

**Issue by marsiancba** [angelozerr/typescript.java#206]
![image](https://user-images.githubusercontent.com/5137942/30163637-ef4dd94c-93d9-11e7-9c0b-b6c15522c3ab.png)

It should insert just `bug1(a)`, as `b` is argument of callback function `a`.

Same problem happens for objects with index signature parameter:

![image](https://user-images.githubusercontent.com/5137942/30163717-4664d3fc-93da-11e7-98a3-2fbb4f66cbde.png)

**Comment by angelozerr** [comment]
Indeed, today I gave not managed that. Just find time to support it.

**Comment by marsiancba** [comment]
It seems that this is implemented in Java with something like Regex. If you can point me to part of the code where this is implemented, I can try to fix that by myself.
